lib/jobs/job_test.go

```diff
@@ -0,0 +1,133 @@
+package jobs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/kosyfrances/rundeck-zabbix/lib/zabbix"
+)
+
+// Test that we can get Rundeck job model document in Yaml format
+// with triggers' information from Zabbix
+func TestMakeJob(t *testing.T) {
+	expected := zabbix.TriggerResults{
+		{
+			Description: "random trigger on dummy-host",
+			Hosts: []zabbix.TriggerHost{
+				{Name: "dummy-host"},
+			},
+		}, {
+			Description: "prefix random trigger on dummy-host2",
+			Hosts: []zabbix.TriggerHost{
+				{Name: "dummy-host2"},
+			},
+		},
+	}
+
+	// mock api call
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		res := struct {
+			zabbix.TriggerResults `json:"result"`
+		}{expected}
+		err := json.NewEncoder(w).Encode(res)
+		if err != nil {
+			t.Fatalf("unable to write response. error: %v", err)
+		}
+	}))
+	defer ts.Close()
+
+	// generate temp file
+	tmpfile, err := ioutil.TempFile("", "testfile.*.yml")
+	if err != nil {
+		t.Fatalf("unable to create temporary file. error: %v", err)
+	}
+
+	defer os.Remove(tmpfile.Name()) // clean up
+	defer tmpfile.Close()           // close file
+
+	a, err := zabbix.CreateClientUsingAPIKey(ts.URL, "fake_zabbix_key")
+	if err != nil {
+		t.Fatalf("process ran with err %v", err)
+	}
+
+	res, err := a.GetTriggersInfo()
+	if err != nil {
+		t.Fatalf("cannot get triggers info.\n%v", err)
+	}
```

Comment:
Hmm ... looks like L55-60 will be forcing this test to depend on the implementations of `CreateClientUsingAPIKey` and `GetTriggersInfo` functions i.e. if they change in future (or refactored) so will this test. Unless for something akin to an integration test, reducing the coupling might be beneficial, particularly if the actual implementation does not depend on them.

Not super important, just something to keep in mind. I do see that's how a couple of other tests are written nvm.